src/tests/test_cases/expected.py

```diff
@@ -4,12 +4,14 @@ class Expectation:
     def __init__(self):
         self.expected = set()
 
-    def expect(self, *args):
+    def expect(self, *args, arg_count=0):
         """Decorator that expects a certain PEP 257 violation."""
         def none(_):
             return None
 
+        # The `arg_count` parameter helps the decorator
+        # with functions that have positional arguments.
         if len(args) == 1:
             return lambda f: (self.expected.add((f.__name__, args[0])) or
-                              none(f()) or f)
+                              none(f(*[None]*arg_count)) or f)
```

Comment:
I didn't like this implementation before, but now it's even harder to read - not your fault, but if you're willing, can you please just write this out as a function definition without the `or` tricks? i.e.,

```py
def decorate(f):
    self.expected.add((f.__name__, args[0]))
    none(f())
    return f
```

At least, I *think* this is how this decorator works. This way, I don't think you need the `none` function, you can just call `f` directly.